cmd/up.go

```diff
@@ -13,48 +13,54 @@ import (
 	"time"
 
 	"github.com/pkg/errors"
-
-	"github.com/alexellis/k3sup/pkg/env"
-	"github.com/openfaas/faasd/pkg"
 	"github.com/sethvargo/go-password/password"
 	"github.com/spf13/cobra"
+	flag "github.com/spf13/pflag"
+
+	"github.com/openfaas/faasd/pkg"
 )
 
+// upConfig are the CLI flags used by the `faasd up` command to deploy the faasd service
+type upConfig struct {
+	// composeFilePath is the path to the compose file specifying the faasd service configuration
+	// See https://compose-spec.io/ for more information about the spec,
+	//
+	// currently, this must be the name of a file in workingDir, which is set to the value of
+	// `faasdwd = /var/lib/faasd`
+	composeFilePath string
+
+	// working directory to assume the compose file is in, should be faasdwd.
+	// this is not configurable but may be in the future.
+	workingDir string
+}
+
+func init() {
+	configureUpFlags(upCmd.Flags())
+}
+
 var upCmd = &cobra.Command{
 	Use:   "up",
 	Short: "Start faasd",
 	RunE:  runUp,
 }
 
-const containerSecretMountDir = "/run/secrets"
-
-func runUp(_ *cobra.Command, _ []string) error {
+func runUp(cmd *cobra.Command, _ []string) error {
 
-	clientArch, clientOS := env.GetClientArch()
-
-	if clientOS != "Linux" {
-		return fmt.Errorf("You can only use faasd on Linux")
+	cfg, err := parseUpFlags(cmd)
+	if err != nil {
+		return err
 	}
-	clientSuffix := ""
```

Comment:
Retain this logic, but use it to set a variable passed into the compose parsing/de-tokenising code.

Reply:
@alexellis it seems like there is a bug here, the `armhf` and `arm64` cases are empty, which means that clientSuffix is empty, is that correct?  it seems like `armhf` should also have the `-armhf` suffix

Notice in this go playground snippet, the second switch (which matches this implementation) does not print anything https://play.golang.org/p/RyDJg4Syinu

Reply:
They are fall-through cases, not empty.

Reply:
But if it doesn't seem to work (your syntax in the playground appears to the be correct one), I wonder how faasd is working on armhf?

Reply:
This is my RPi with version 0.8.2 running:

```
pi@faasd-pi:~ $ sudo ctr c ls
CONTAINER            IMAGE                                                 RUNTIME                  
basic-auth-plugin    docker.io/openfaas/basic-auth-plugin:0.18.10-armhf    io.containerd.runc.v2    
gateway              docker.io/openfaas/gateway:0.18.8-armhf               io.containerd.runc.v2    
nats                 docker.io/library/nats-streaming:0.11.2               io.containerd.runc.v2    
prometheus           docker.io/prom/prometheus:v2.14.0                     io.containerd.runc.v2    
queue-worker         docker.io/openfaas/queue-worker:0.9.0                 io.containerd.runc.v2    
pi@faasd-pi:~ $ sudo ctr t ls
TASK                 PID      STATUS    
basic-auth-plugin    11900    RUNNING
nats                 12005    RUNNING
prometheus           12127    RUNNING
gateway              12250    RUNNING
queue-worker         12365    RUNNING
pi@faasd-pi:~ $ 
```

Reply:
I suspect it's working because the actual value is `armv7l` and not `armhf`. So you have found a bug there, which I guess a unit test on my behalf would have found too.

Can you see if this issue (incorrect use of fall-through case statements) exists in arkade or faas-cli too?

Reply:
@LucasRoesler do you need more direction on this?

Just fix up my incorrect usage of case statements, and move over to `case "value1", "value2":` instead.

Reply:
ok, this is already pushed here https://github.com/openfaas/faasd/pull/76/files#diff-5ea88327532ebc9c00c6203cbd1efde0R326